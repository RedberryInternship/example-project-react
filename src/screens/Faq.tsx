import React, {useState, ReactElement, useEffect} from 'react'
import {View, StyleSheet, ScrollView} from 'react-native'
import i18next from 'i18next'

import {NavigationScreenProp} from 'react-navigation'

// components
import {BaseHeader, FaqListItem} from 'components'

// utils
import {Colors, Ajax, Defaults, getLocaleText} from 'utils'
import {LocaleStringObject} from 'allTypes'

type FAQResponseType = {
  question: LocaleStringObject
  answer: LocaleStringObject
}

let FAQStatic: FAQResponseType[] = []

type FaqPropsType = {
  navigation: NavigationScreenProp<any, any>
}

const Faq = ({navigation}: FaqPropsType): ReactElement => {
  const [activeFaq, setActiveFaq] = useState<number>(1)
  const [faqs, setFaqs] = useState(FAQStatic)

  useEffect(() => {
    getFAQ()
  }, [])

  const getFAQ = async (): Promise<void> => {
    if (FAQStatic.length === 0) {
      try {
        const res = await Ajax.get('/faq')
        setFaqs(res.faq)
        FAQStatic = res.faq
      } catch (error) {
        Defaults.dropdown?.alertWithType(
          'error',
          i18next.t('dropDownAlert.generalError'),
        )
      }
    }
  }

  return (
    <View style={styles.container}>
      <BaseHeader
        title={'faq.frequentlyAskedQuestions'}
        onPressLeft={navigation.navigate.bind(Faq, 'MainDrawer')}
      />
      <ScrollView style={styles.scrollViewStyle}>
        {faqs.map((el, ind) => (
          <FaqListItem
            key={ind}
            number={ind + 1}
            question={getLocaleText(el.question)}
            answer={getLocaleText(el.answer)}
            activeFaq={activeFaq}
            setActiveFaq={setActiveFaq}
          />
        ))}
      </ScrollView>
    </View>
  )
}

export default Faq

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.primaryBackground,
  },
  scrollViewStyle: {
    marginTop: 32,
  },
})
