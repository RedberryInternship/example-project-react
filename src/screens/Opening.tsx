import React from 'react'
import {StyleSheet, View} from 'react-native'
import {Colors, GNOME} from 'utils'
import {BaseText} from 'components'

const Opening = () => {
  return (
    <View style={styles.container}>
      <BaseText style={styles.text}>Welcome to E-Space</BaseText>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: Colors.primaryBackground,
  },
  text: {
    fontFamily: GNOME.HELV_HVEX,
  },
})

export default Opening
