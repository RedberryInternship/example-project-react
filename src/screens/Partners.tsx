import React, {ReactElement, useEffect, useState} from 'react'
import {View, StyleSheet, SafeAreaView, Image} from 'react-native'

// components
import {BaseHeader, FetchedDataRenderer} from 'components'

// utils
import {Colors, Ajax, Const} from 'utils'

type PartnersResponseType = {
  name: string
  image: string
}

const Partners = ({navigation}: any): ReactElement => {
  const getPartners = async (): Promise<void> => {
    const res = await Ajax.get('/partners')
    return res.partners
  }

  return (
    <View style={styles.container}>
      <BaseHeader
        title={'partners.partnerOrganizations'}
        onPressLeft={navigation.navigate.bind(Partners, 'MainDrawer')}
      />
      <View style={styles.partnersInnerContainer}>
        <FetchedDataRenderer
          property={'Partners'}
          onItemRender={(val: PartnersResponseType, index): ReactElement => (
            <View key={index} style={styles.partnerImageContainer}>
              <Image
                source={{uri: val.image}}
                style={{width: 80, height: 40}}
                resizeMode={'contain'}
              />
            </View>
          )}
          fetchData={getPartners}
          updateAlways={true}
        />
      </View>
      <SafeAreaView />
    </View>
  )
}

export default Partners

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.primaryBackground,
  },
  partnersInnerContainer: {
    flex: 1,
    backgroundColor: Colors.secondaryGrey,
    marginTop: 32,
    marginHorizontal: 16,
    marginBottom: 16,
    borderRadius: 10,
    padding: 32,
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  partnerImageContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 32,
    width: (Const.Width - 32) / 3,
    height: 80,
  },
})
