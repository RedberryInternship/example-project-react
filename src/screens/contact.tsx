import React, { useState } from 'react';
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  Image,
  KeyboardAvoidingView,
  Platform,
  Linking,
  Alert,
  StatusBar
} from 'react-native';


import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

import { useTranslation } from 'react-i18next'
import { useSafeArea } from 'react-native-safe-area-context';

// components
import {
  BaseHeader,
  BaseButton,
  ContactListItem
} from '../components';

// utils
import {
  Colors,
  Const,
  Defaults
} from '../utils';

const contact = ({ navigation }: any) => {


  const { t } = useTranslation();
  const [message, setMessage] = useState();
  const insets = useSafeArea();

  const listItems = Const.ContactListFields
    .map((el, key) => {
      return <ContactListItem
              key={el.type}
              image={el.image}
              name={el.name}
              value={contactInfos[key]}
              onPress={outgoingLinkMethods[el.type]}
            />
  });


  const sendMessage = (msg: string) => {
    Alert.alert("", msg, [{ text: "Got It!", onPress: () => navigation.goBack() }]);
  }


  return (
    <View style={[styles.container,{paddingBottom : insets.bottom + 16}]}>
      <BaseHeader
        title={'contact.contact'}
        onPressLeft={() => navigation.goBack() } />

      <KeyboardAwareScrollView
        keyboardShouldPersistTaps="handled"
        enableAutomaticScroll={true}
        extraScrollHeight={Platform.select({ ios: -300, android: 150 })}
        showsVerticalScrollIndicator={false}
        enableResetScrollToCoords={true}
        contentContainerStyle={{ flex: 0 }}
        overScrollMode={"always"}
        extraHeight={Platform.select({ ios: 500, android: 75 })}
        // keyboardDismissMode={"on-drag"}
        resetScrollToCoords={{ x: 0, y: 0 }} >
        <View style={styles.contactItemsContainer}>
          {listItems}
        </View>

        <View style={styles.messageContainer}>
          <Text style={styles.messageTitle}>{t("contact.message")}</Text>
          <Image source={require("../../assets/images/icons/mail.png")} style={styles.messageIcon} />
          <TextInput
            multiline={true}
            style={styles.message}
            onChangeText={(text) => setMessage(text)}
            numberOfLines={4}
          />
        </View>
      </KeyboardAwareScrollView>



      <KeyboardAvoidingView
        behavior="padding"
        // contentContainerStyle={{ flex: 1 }} 
        keyboardVerticalOffset={Platform.OS === "ios" ? 16 : StatusBar.currentHeight}>
        <BaseButton
          onPress={() => sendMessage(message)}
          text="save"
          image={require("../../assets/images/icons/arrow_left.png")}
          style={{ marginTop: 0, marginBottom: 16 }}
          isImageRight={true} />
      </KeyboardAvoidingView>
    </View>
  );
}

export default contact;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.primaryBackground,
    // paddingBottom: 32
  },
  contactItemsContainer: {
    backgroundColor: Colors.secondaryGray,
    marginHorizontal: 16,
    borderRadius: 8,
    paddingHorizontal: 16,
    paddingTop: 32,
    marginTop: 32
  },
  messageContainer: {
    marginHorizontal: 16,
    marginVertical: 32,
    position: "relative",
  },
  messageTitle: {
    color: Colors.primaryGray,
    marginBottom: 16
  },
  messageIcon: {
    position: "absolute",
    zIndex: 1,
    width: 24,
    height: 24,
    top: 40,
    left: 8
  },
  message: {
    height: 200,
    backgroundColor: Colors.black,
    borderRadius: 8,
    color: Colors.primaryWhite,
    paddingLeft: 40,
    paddingRight: 16,
    paddingTop: 10,
    textAlignVertical: "top",

  }
});

const contactInfos = [
  "საირმის ქუჩა 11ო",
  "+995 591 93 50 80",
  "gela@espace.ge",
  "e-space",
  "www.espace.ge"
];


const outgoingLinkMethods : any = {

  "address": () => {

    const mapsInfo = {
      scheme: Platform.select({ android: 'geo:0,0?q=', ios: 'maps:0,0?q=' }),
      latitude: 41.707204,
      longitude: 44.784487,
      label: 'E-space'
    };

    const mapsUrl = `${mapsInfo.scheme}@${mapsInfo.latitude},${mapsInfo.longitude}( ${mapsInfo.label} )`;

    Linking.openURL(mapsUrl);
  },

  "phone": () => {
    Linking.canOpenURL(`tel:591935080`).then(supported => {
      if (supported) {
        Linking.openURL(`tel:591935080`);
      }
      else {
        Defaults.dropdown.alertWithType("error", "Error", "Something Went Wrong While Calling...");
      }
    })
      .catch(err => {
        Defaults.dropdown.alertWithType("error", "Error", "Something Went Wrong While Calling...");
        console.log(err)
      });
  },

  "eMail": () => {
    Linking.canOpenURL(`mailto:gela@espace.ge`).then(supported => {
      if (supported) {
        Linking.openURL(`mailto:gela@espace.ge?subject=e-space`);
      }
    })
      .catch(err => {
        Defaults.dropdown.alertWithType("error", "Error", "Something Went Wrong While Opening Email...");
        console.log(err)
      });
  },

  "facebookPage": () => {
    Linking.canOpenURL('fb://group/272061007052173')
      .then(supported => {
        if (supported) {
          Linking.openURL('fb://group/272061007052173');
        }
        else {
          Linking.openURL('https://www.facebook.com/groups/272061007052173/');
        }
      })
      .catch(err => {
        Defaults.dropdown.alertWithType("error", "Error", "Something Went Wrong While Opening Facebook...");
        console.log(err)
      });
  },

  "webPage": () => {
    Linking.canOpenURL('http://e-space.ge/')
      .then(supported => {
        if (supported) {
          Linking.openURL('http://e-space.ge/');
        }
        else {
          Defaults.dropdown.alertWithType("error", "Error", "Something Went Wrong while Opening Web Page...");
        }
      })
      .catch(err => {
        Defaults.dropdown.alertWithType("error", "Error", "Something Went Wrong while Opening Web Page...");
        console.log(err)
      });
  }
}

