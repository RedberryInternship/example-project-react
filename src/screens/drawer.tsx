import React from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Alert,
  TouchableOpacity,
  Text
} from 'react-native';


// import components
import {
  BaseButton,
  BaseLocaleButton,
  DrawerTextFieldItem,
  BaseUserAvatarWithLabel
} from '../components';

// import utils
import { Const, Colors, NavigationActions } from '../utils';
import { useTranslation } from 'react-i18next';

const drawer = ({ navigation } : any) => {

  const {  i18n } = useTranslation();

  const $isUserAuthorized = true;
  let $drawerListFields = null;
  let $drawerContent = null;

  if (!$isUserAuthorized) {
    $drawerListFields = Const.DrawerFieldsBeforeAuthorization.map((Field, ind) => {
      return <DrawerTextFieldItem
        key={ind}
        onPress={() => navigation.navigate(Field.route) }
        text={Field.text}
        image={Field.image}
        />;
    });

    $drawerContent = 
      <>
        <View style={{ flex: 0 }}>
          <BaseButton image={require("../../assets/images/icons/user.png")}
            onPress={() => undefined}
            text={'home.authorization'}
            style={styles.drawerAuthBtn} />

          {$drawerListFields}
        </View>

        <View style={{ flex: 0 }}>
          <DrawerTextFieldItem
            onPress={() => {Alert.alert("asfas")}}
            text={'drawer.terms_and_conditions'}
            image={require("../../assets/images/icons/green-tick.png")} />
  
        </View>
      </>;
  }
  else {

    $drawerListFields = Const.DrawerFieldsAfterAuthorization.map(Field => {
      return <DrawerTextFieldItem
        onPress={() => navigation.navigate(Field.route) }
        text={Field.text}
        image={Field.image}
        badge={ Field.route === 'notifications' ? 1 : 0 } />
    })

    $drawerContent = 
      <View>
        <BaseUserAvatarWithLabel onPress={()=>{Alert.alert("change icon")}} />
        {$drawerListFields}
      </View>;
  }

  return (
    <SafeAreaView style={styles.safeAreaViewContainer}>
      <ScrollView 
        bounces={false}
        style={{ flex: 1 }} 
        contentContainerStyle={{ flex: 1, justifyContent: "space-between" }}>

        {$drawerContent}

        <View style={{flexDirection:"row", justifyContent:"space-between", alignItems:"center"}}>  
          <BaseLocaleButton
            onPress={() => { i18n.changeLanguage(i18n.language === 'ka' ? 'en' : 'ka') }}
            text={i18n.language === 'ka' ? 'Eng' : 'Ka'}
            style={styles.localeButton} 
          />
          <TouchableOpacity onPress={() =>{Alert.alert("sdf")}} >
              <Text style={{marginRight:24, color:"white"}}>Log out</Text>
          </TouchableOpacity>
        </View>
        
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  safeAreaViewContainer: {
    flex: 1,
    borderBottomLeftRadius: 24,
    borderTopLeftRadius: 24,
    backgroundColor: Colors.primaryBackground
  },
  container: {
    flex: 1,
    backgroundColor: Colors.primaryBackground
  },
  drawerAuthBtn: {
    width: Const.Width - 120,
    marginTop: 10,
    marginBottom: 60
  },
  localeButton: {
    marginLeft: 24,
    marginTop: 20,
    marginBottom: 20
  }
});

export default drawer;