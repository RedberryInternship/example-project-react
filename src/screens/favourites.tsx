import React from 'react';
import {
  ScrollView,
  SafeAreaView,
  StyleSheet,
  Alert
} from 'react-native';


import {
  BaseHeader,
  FavouriteChargerListItem
} from '../components';

const favourites = ({ navigation }: any) => {


  return (
    <SafeAreaView>
      <BaseHeader
        title={'favourites.favourites'}
        onPressLeft={() => navigation.goBack()} />

      <ScrollView style={styles.container} >

        <FavouriteChargerListItem
          title='თბილისი მოლი'
          address='საიათნოვას ქუჩა'
          turnon={() => Alert.alert("this is a turnon!")}
          deleteItem={() => Alert.alert("this deletes the item!")} />

        <FavouriteChargerListItem
          title='ლილო მოლი'
          address='ლუბლიანას ქუჩა'
          turnon={() => Alert.alert("this is a turnon!")}
          deleteItem={() => Alert.alert("this deletes the item!")} />


      </ScrollView>
    </SafeAreaView>
  )
}

export default favourites;

const styles = StyleSheet.create({
  container: {
    marginTop: 64,
    paddingBottom: 32
  }
});