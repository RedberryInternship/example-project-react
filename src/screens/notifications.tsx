import React from 'react'
import {ScrollView, StyleSheet, View, SafeAreaView} from 'react-native'

// components
import {BaseHeader, NotificationListItem} from 'components'

// utils
import {Colors} from 'utils'

const notifications = ({navigation}: any) => {
  const notificationItems = notificationsList.map(el => {
    return <NotificationListItem key={el.title} {...el} />
  })

  return (
    <View style={styles.container}>
      <BaseHeader
        title={'notifications.notifications'}
        onPressLeft={() => navigation.navigate('MainDrawer')}
      />
      <ScrollView style={styles.scrollViewContainer}>
        {notificationItems}
      </ScrollView>
      <SafeAreaView />
    </View>
  )
}

export default notifications

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.primaryBackground,
  },
  scrollViewContainer: {
    marginTop: 16,
  },
})

const notificationsList = [
  {
    title: 'სისტემური განახლება',
    description:
      'გაცნობებთ, რომ მიმდინარეობს სისტემური განახლება! პანიკის გარეშე...',
    date: '12/11/2020',
  },
  {
    title: 'რელევანტური ინფორმაცია',
    description:
      'გაცნობებთ, რომ მიმდინარეობს სისტემური განახლება! პანიკის გარეშე...',
    date: '12/11/2020',
  },
  {
    title: 'არარელევანტური ინფორმაცია',
    description:
      'გაცნობებთ, რომ მიმდინარეობს სისტემური განახლება! პანიკის გარეშე...',
    date: '12/11/2020',
  },
  {
    title: 'კოჰაბიტაცია',
    description:
      'გაცნობებთ, რომ მიმდინარეობს სისტემური განახლება! პანიკის გარეშე...',
    date: '12/11/2020',
  },
  {
    title: 'მამათმავლობა',
    description:
      'გაცნობებთ, რომ მიმდინარეობს სისტემური განახლება! პანიკის გარეშე...',
    date: '12/11/2020',
  },
  {
    title: 'ველისციხე',
    description:
      'გაცნობებთ, რომ მიმდინარეობს სისტემური განახლება! პანიკის გარეშე...',
    date: '12/11/2020',
  },
]
