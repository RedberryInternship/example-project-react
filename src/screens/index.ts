import Home from "./tabNavigation/home"
import ChargerWithCode from "./tabNavigation/charger/chargerWithCode"
import ChargerDetail from "./tabNavigation/charger/chargerDetail"
import NotAuthorized from "./tabNavigation/charger/notAuthorized"
import ChooseChargeMethod from "./tabNavigation/charger/chooseChargeMethod"
import Charging from "./tabNavigation/charger/charging"
import ChoosingCard from "./tabNavigation/charger/choosingCard"

import Plashka from "./plashka"
import Drawer from "./drawer"
import Auth from "./authentication/auth"
import ForgotPassword from "./authentication/forgotPassword"
import Registration from "./authentication/registration"
import SetNewPasswords from "./authentication/setNewPasswords"
import Settings from "./drawerMenuOptions/settings";
import ProfileChange from './drawerMenuOptions/profileChange';

import Favorites from './favourites';
import Faq from './faq';
import Tariffs from './tariffs';

export {
    Home,
    Auth,
    Plashka,
    ChargerWithCode,
    Drawer,
    ForgotPassword,
    Registration,
    SetNewPasswords,
    Settings,
    ProfileChange,
    ChargerDetail,
    NotAuthorized,
    ChooseChargeMethod,
    Charging,
    Favorites,
    Faq,
    ChoosingCard,
    Tariffs
}
