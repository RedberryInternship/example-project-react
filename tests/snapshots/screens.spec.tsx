import React from 'react'
import { Opening } from 'screens'
import { render } from '@testing-library/react-native';

it('Snapshorts auth screen', () => {
  const authScreen = render(<Opening />).toJSON();
  expect(authScreen).toMatchSnapshot();
})
