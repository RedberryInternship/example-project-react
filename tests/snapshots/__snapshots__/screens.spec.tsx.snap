// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshorts auth screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#11222D",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../src/assets/images/logo.png",
      }
    }
    style={
      Object {
        "width": 100,
      }
    }
  />
  <Text
    allowFontScaling={false}
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 13,
          "letterSpacing": 0.2,
          "textAlignVertical": "center",
        },
        Object {
          "fontFamily": "HelveticaNeue-Bold",
          "marginBottom": 50,
        },
        Object {
          "fontFamily": "HelveticaNeue-Bold",
          "fontSize": 13,
          "letterSpacing": 0.2,
          "lineHeight": 13,
        },
      ]
    }
    testID="title"
  >
    Welcome to E-Space
  </Text>
</View>
`;
